File: helixtrajectorycpp/src/TrajectorySample.cpp
#include "TrajectorySample.h"

namespace helixtrajectory {

    TrajectorySample::TrajectorySample(double x, double y, double heading)
            : x(x), y(y), heading(heading) {
    }

    TrajectorySample::~TrajectorySample() {
    }
}

File: helixtrajectoryj/src/main/java/org/team2363/helixtrajectory/TrajectorySample.java
package org.team2363.helixtrajectory;

public abstract class TrajectorySample {

    public final double x;
    public final double y;
    public final double heading;

    protected TrajectorySample(double x, double y, double heading) {
        this.x = x;
        this.y = y;
        this.heading = heading;
    }
}